### DISCUSSION TITLE
Creaating divs on the fly

### DISCUSSION THREAD

**Discussion by craigbass76** [mkdocs/mkdocs#3042] [Q&A]
```
:::defs

- First
    - blah
- Second
    - lahblah
- Third
    - blahblahblah

:::
```
I'm using pandoc (which I've just learned has it's own flavor of markdown) and it renders the above code as:
```
<div class="defs">
<ul>
<li>First
<ul>
<li>blah</li>
</ul></li>
<li>Second
<ul>
<li>lahblah</li>
</ul></li>
<li>Third
<ul>
<li>blahblahblah</li>
</ul></li>
</ul>
</div>
```

I guess MkDocs uses python-flavored Markdown. Here's what my source looks like in MkDocs:
```
<p>:::defs</p>
<ul>
<li>First<ul>
<li>blah</li>
</ul>
</li>
<li>Second<ul>
<li>lahblah</li>
</ul>
</li>
<li>Third<ul>
<li>blahblahblah</li>
</ul>
</li>
</ul>
```

Is there any quick way to create a div with a class, besides hand coding the HTML? I'm trying to do two different things with the same `.md` files.

**Comment by oprypin** [comment]
This is the first time I see this `:::` syntax. Not aware of anything like this.

The closest thing is indeed writing HTML manually
but at least you can write Markdown within the HTML
if you enable the `md_in_html` extension
https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#markdown-in-html

**Comment by facelessuser** [comment]
There is a solution being worked that should allow something _like_ this (but not exactly like this), I just need to set aside some time to complete the work: https://github.com/facelessuser/pymdown-extensions/pull/1777.

**Comment by gledos** [comment]
May be use CommonMark style.

```markdown
<div class="defs">

- First
    - blah
- Second
    - lahblah
- Third
    - blahblahblah

</div>
```

![圖片](https://user-images.githubusercontent.com/23720620/204075766-37f01d00-2765-47f4-aa7d-bee8f2defc44.png)

https://spec.commonmark.org/0.30/#example-188

**Reply by gledos** [reply]
I tested it and it doesn't seem to work...

May be only use the `md_in_html` extension.
